src/librustc_trans/back/write.rs

```diff
@@ -1394,13 +1394,17 @@ fn start_executing_work(tcx: TyCtxt,
                         allocator_config: Arc<ModuleConfig>)
                         -> thread::JoinHandle<Result<CompiledModules, ()>> {
     let coordinator_send = tcx.tx_to_llvm_workers.clone();
+    let sess = tcx.sess;
     let mut exported_symbols = FxHashMap();
-    exported_symbols.insert(LOCAL_CRATE, tcx.exported_symbols(LOCAL_CRATE));
-    for &cnum in tcx.crates().iter() {
-        exported_symbols.insert(cnum, tcx.exported_symbols(cnum));
+    if sess.lto() != Lto::No {
+        exported_symbols.insert(LOCAL_CRATE, tcx.exported_symbols(LOCAL_CRATE));
+        if sess.lto() != Lto::ThinLocal {
+            for &cnum in tcx.crates().iter() {
+                exported_symbols.insert(cnum, tcx.exported_symbols(cnum));
+            }
+        }
     }
```

Comment:
Could you turn this whole block into a `match tcx.sess.lto() { ... }` instead of two nested `if`s? It's a bit more verbose but it makes it easier to read.